src/Lib/Modes/PathEditMode.vala

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public override void mode_end () {
+        // Hide the path layer and show nobs.
+        view_canvas.toggle_layer_visibility (ViewLayers.ViewLayer.NOBS_LAYER_ID, true);
+        view_canvas.toggle_layer_visibility (ViewLayers.ViewLayer.PATH_LAYER_ID, false);
+    }
+
+    public override Gdk.CursorType? cursor_type () {
+        return Gdk.CursorType.CROSSHAIR;
+    }
+
+    public override bool key_press_event (Gdk.EventKey event) {
+        return false;
+    }
+
+    public override bool key_release_event (Gdk.EventKey event) {
+        return false;
+    }
+
+    public override bool button_press_event (Gdk.EventButton event) {
+        // Everytime the user presses the mouse button, a new point needs to be created and added to the path.
+        Akira.Geometry.Point point = Akira.Geometry.Point (event.x, event.y);
+
+        if (first_point.x == -1) {
+            first_point = point;
+            update_view ();
+            return false;
+        }
+
+        // This calculates the position of the new point relative to the first point.
+        point.x -= first_point.x;
+        point.y -= first_point.y;
+
+        // Add the new points to the drawable and path
+        add_point_to_path (point);
+
+        // To calculate the new center of bounds of rectangle,
+        // Move the center to point where user placed first point. This is represented as (0,0) internally.
+        // Then translate it to the relative center of bounding box of path.
+        var bounds = instance.components.path.calculate_extents ();
+        double center_x = first_point.x + bounds.center_x;
+        double center_y = first_point.y + bounds.center_y;
+
+        instance.components.center = new Lib.Components.Coordinates (center_x, center_y);
+        instance.components.size = new Lib.Components.Size (bounds.width, bounds.height, false);
+
+        // Update the component.
+        view_canvas.items_manager.item_model.mark_node_geometry_dirty_by_id (instance.id);
+        view_canvas.items_manager.compile_model ();
+
+        update_view ();
+
+        return true;
+    }
+
+    private void add_point_to_path (Geometry.Point point, int index = -1) {
+
+        var old_path_points = instance.components.path.data;
+        Geometry.Point[] new_path_points = new Geometry.Point[old_path_points.length + 1];
+
+        index = (index == -1) ? index = old_path_points.length : index;
+
+        for (int i = 0; i < index; ++i) {
+            new_path_points[i] = old_path_points[i];
+        }
+
+        new_path_points[index] = point;
+
+        for (int i = index + 1; i < old_path_points.length + 1; ++i) {
+            new_path_points[i] = old_path_points[i - 1];
+        }
+
+        instance.components.path = new Lib.Components.Path.from_points (new_path_points);
+    }
+
+    /*
+     * Recalculates the extents and updates the ViewLayerPath
+     */
+    private void update_view () {
```

Comment:
This should be done by overloading the existing "update ()" method in the corresponding drawable. (see ViewLayersHover.update () )

Reply:
@mbfraga I tried to do this similar to **ViewLayerHover** by passing the drawable to **ViewLayerPath** and getting bounds and points from it.
But that leads to a bug. Placing points in a zig zag manner outside the current bounds causes the points to shift from their original position. It will take some time to figure why that's happening. 
So for now, I will manually calculate the bounds, pass it to a ViewLayerPath to update its variables and then call update from there.

Reply:
ok, I can take a look at this later. no problem.